creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+TEMP_STUDY_BUCKET = "kf-api-study-creator-ingest-test"
+boto3.setup_default_session(profile_name=AWS_PROFILE)
+
+
+class StudyGenerator(object):
+
+    def __init__(
+        self,
+        data_dir=DATA_DIR,
+        ingest_package_dir=INGEST_PKG_DIR,
+        study_id="SD_ME0WME0W",
+        sequencing_centers=None,
+        total_specimens=10,
+    ):
+        self.data_dir = data_dir
+        self.study_id = study_id
+        if not sequencing_centers:
+            sequencing_centers = [
+                {
+                    "kf_id": "SC_DGDDQVYR",
+                    "name": "Broad Institute",
+                }
+            ]
+        self.sequencing_centers = sequencing_centers
+        self.total_specimens = total_specimens
+        self.source_gf_extensions = [".cram", ".crai", ".cram.md5"]
+        self.logger = logging.getLogger(type(self).__name__)
+
+        # Initialize ingest - use default settings (KF Data Service)
+        self.ingest_pipeline = DataIngestPipeline(
+            ingest_package_dir,
+            ingest_settings.load().TARGET_API_CONFIG,
+        )
+        # KF ID Prefixes
+        self.id_prefixes = {
+            "family": "FM",
+            "participant": "PT",
+            "biospecimen": "BS",
+            "genomic_file": "GF",
+        }
+
+    @property
+    def study_bucket(self):
+        """ Return study bucket from study KF ID """
+
+        return TEMP_STUDY_BUCKET
+        # return f"kf-study-us-east-1-prd-{study_id.lower().replace('_', '-')}"
+
+    def ingest_study(self, initialize=False, create_files=False):
+        """ 
+        Entrypoint. Ingest study data files into Data Service 
+
+        :param initialize: Create the study and sequencing centers in the data
+        service
+        :type initialize: bool
+        :param create_files: Create the study data files if True, otherwise
+        read them from disk
+        :type create_files: bool
+        """
+
+        # Create study, sequencing center info
+        if initialize:
+            self._initialize_study()
+
+        # Create new or read in existing study data files
+        kf_ids = self.generate_files(create=create_files)
+
+        # Ingest study data files into Data Service
+        self.ingest_pipeline.run()
+
+        # Ensure all kf_ids were loaded
+        for endpoint, kfids in kf_ids.items():
+            self._assert_load(endpoint, kfids)
+
+    def generate_files(self, create=False):
+        """
+        Generate clinical and genomic data files for a test study
+
+        :param create: Create new data files or read existing files
+        :type create: bool
+        """
+        ra.seed(0)
+        self.logger.info(
+            f"Creating clinical and genomic files for {self.study_id}"
+        )
+
+        # Create clinical and genomic data files
+        bio_df = self._create_bio_manifest(create=create)
+        gf_df = self._create_gfs(create=create)
+
+        # Upload unharmonized and harmonized gfs to S3
```

Comment:
I added an upload method so we don't have to manually upload the files. It was getting tedious so decided to do it via code after all.